### PR TITLE
letsencrypt: Update to Certbot 2.11.0/Alpine 3.20/Python 3.12

### PR DESCRIPTION
**Pull request by agners**
Update to the latest version of Certbot and ACME 2.11.0. While at it, also use the latest base container with Alpine 3.20 with Python 3.12.

## Summary by CodeRabbit

- **New Features**
  - Upgraded to version 5.2.0, enhancing overall performance and compatibility.

- **Improvements**
  - Updated Certbot/ACME to version 2.11.0, improving security and functionality.
  - Upgraded base images to Python 3.12 and Alpine 3.20 for better performance across architectures.
  - Updated several dependency versions, including Cloudflare and Cryptography, to enhance security and compatibility.

- **Deprecations**
  - Removed support for Google Domains due to lack of ACME support from the new operator, Squarespace.

### CODE REVIEW

**Comment by agners** [comment]
It seems that ClouDNS dependency restriction block upgrade here:
https://github.com/inventage/certbot-dns-cloudns/blob/master/pyproject.toml#L27-L28

@stephenjust @tedski  you've been previously involved in adding ClouDNS support, can you maybe tackle this? I guess loosing the restrictions in `certbot-dns-cloudns` might be all that is needed :thinking:

File: letsencrypt/build.yaml [hunk 1]
```diff
@@ -1,27 +1,27 @@
...
 args:
-  CLOUDFLARE_VERSION: 2.9.11
-  CRYPTOGRAPHY_VERSION: 41.0.5
-  CERTBOT_VERSION: 2.7.4
-  CERTBOT_DNS_AZURE_VERSION: 2.4.0
+  CLOUDFLARE_VERSION: 2.19.4
+  CRYPTOGRAPHY_VERSION: 43.0.1
+  CERTBOT_VERSION: 2.11.0
+  CERTBOT_DNS_AZURE_VERSION: 2.5.0
   CERTBOT_DNS_CLOUDNS_VERSION: 0.6.0
```
Comment (tedski) [inline letsencrypt/build.yaml:16]: bump to 0.7.0 which depends on acme/certbot 2.11.0

File: letsencrypt/Dockerfile [hunk 2]
```diff
@@ -53,7 +52,6 @@ RUN \
         certbot==${CERTBOT_VERSION} \
         certbot-dns-azure==${CERTBOT_DNS_AZURE_VERSION} \
         certbot-dns-cloudflare==${CERTBOT_VERSION} \
-        certbot-dns-cloudns==${CERTBOT_DNS_CLOUDNS_VERSION} \
```
Comment (tedski) [inline letsencrypt/Dockerfile:56]: This can be included with the newly available version 0.7.0 of certbot-dns-cloudns.

**Review by tedski — APPROVED** [review]
lgtm

**Review by sairon — APPROVED** [review]
LGTM, thanks @agners, and thanks @tedski for unblocking this!

**Pull request merged**